clustered/client/src/main/java/org/ehcache/clustered/client/internal/store/StrongServerStoreProxy.java

```diff
@@ -178,16 +160,28 @@ private void reconnectListener(ClusterTierReconnectMessage reconnectMessage) {
     }
   }
 
-  private void awaitOnLatch(CountDownLatch countDownLatch) throws InterruptedException {
-    int totalAwaitTime = 0;
-    int backoff = 1;
-    while (!countDownLatch.await(backoff, TimeUnit.SECONDS)) {
-      totalAwaitTime += backoff;
-      backoff = (backoff >= 10) ? 10 : backoff * 2;
-      LOGGER.debug("Waiting for the server's InvalidationDone message for {}s, backing off {}s...", totalAwaitTime, backoff);
-    }
-    if (!entity.isConnected()) {
-      throw new IllegalStateException("Cluster tier manager disconnected");
+  private void awaitOnLatch(CountDownLatch countDownLatch, long timeoutNanos) throws TimeoutException {
+    boolean interrupted = Thread.interrupted();
+    try {
+      while (true) {
+        try {
+          if (countDownLatch.await(timeoutNanos, TimeUnit.NANOSECONDS)) {
```

Comment:
This doesn't work. If the goal is to wait of `timeoutNanos` even if we are interrupted (and I don't think it it really useful), them in the `catch(InterruptedException e)` we should reduce `timeoutNanos` of what we waited so far.

Reply:
Fixed.